pints/_mcmc/__init__.py

```diff
@@ -774,16 +830,30 @@ def run(self):
             if self._log_to_screen:
                 print(halt_message)
 
-        # Store generated chains in memory
-        if self._chains_in_memory:
-            self._samples = samples
-
         # Store evaluations in memory
         if self._evaluations_in_memory:
             self._evaluations = evaluations
 
+        # Inverse transform to model space if transform is provided
+        if self._chains_in_memory:
+            if self._transform:
+                n_c, n_s, n_p = samples.shape
+                samples_user = np.zeros((n_c, n_s, n_p))
+                for c in range(n_c):
+                    for s in range(n_s):
+                        samples_user[c, s, :] = \
+                            self._transform.to_model(samples[c, s, :])
```

Comment:
If this is going to be the standard use we should probably make it so that transformations can do this with 1 numpy command instead of in a nested for loop (much slower!)

Reply:
Ha!!! I did that and removed it... Because that'd involve in the ComposeTransforms checking whether it's a single parameter or a set of parameters... And that in the Transform docs, it'd become slightly tricky, because we have to tell users whoever try to implement their own that the `Transform.to_xxx()` can take either `(n_parameters,)` 1D array or `(n_sets, n_parameters)` 2D array...

Reply:
OK, could you maybe run an O(10000) samples MCMC chain and do a log transform both ways, just to see if there's an appreciable difference?
We could always do a `many_to_model()` method or something in the transformation, with a slow default implementation in the base class?

Reply:
>We could always do a many_to_model() method or something in the transformation, with a slow default implementation in the base class?
Yup, don't mind doing this one, if you think that makes sense too!

Reply:
Done. It's called `multiple_to_model()` (instead of "many", so it's just anything more than one...).